cparser/processor.js

```diff
@@ -81,7 +81,7 @@ module.exports = that = {
 			var missingFuncs = regexParser.getMissingDeclarations(cleanText);
 
 			var prototypesStr = missingFuncs.join('\n') + '\n'
-				+ '#line ' + linesBeforeInjection + '\n';
+				+ '#line ' + linesBeforeInjection + ' "' + inputFile + '"\n';
```

Comment:
I was thinking the same thing but then when compiling `.ino` files would show up as short names i.e. `src/input.ino` but all other referenced files would have full paths like `/workspace/lib/foo/src/bar.cpp`. So when parsing errors we would get two different styles of paths. This is why I'm setting full paths here so then normalization would be handled [after compilation(https://github.com/spark/buildpack-base/blob/master/bin/run#L30).